server/src/lib/server.rs

```diff
@@ -93,6 +96,12 @@ pub(crate) struct InstanceContext {
         Mutex<BTreeMap<Uuid, Arc<propolis::block::CrucibleBackend>>>,
 }
 
+#[derive(Debug, Clone)]
+pub struct InstanceMetrics {
+    pub(crate) producer_registry: Arc<Mutex<Option<ProducerRegistry>>>,
+    pub(crate) pso: Arc<Mutex<Option<PropStatOuter>>>,
```

Comment:
It seems to me these `Arc<Mutex<...>>` are a bit redundant as both `ProducerRegistry` and `PropStatOuter` already contain `Arc`'s themselves and have `Clone` impls that do the right thing if you just passed them around directly.

I would also just remove the `Option`'s here and instead wrap `InstanceMetrics` as a whole with `Option` in `Context` below.

Reply:
Yes, I agree on the redundant part.

I tried to get things to work without them, but either I could not figure it out, or something about the way the `server::Context` was passed around was preventing just a straight `Option<T>` here.  I'll give it another try and if I can't figure it out then I'll come back for help :)

Reply:
Yea, the ownership of things gets complicated. Feel free to ping me if you run into anything gnarly.

Reply:
The latest commit does improve this, (thanks for your help out of band) and I was also able to keep `producer_registry` only inside `instance_ensure()` so things got simpler as well.